### PR TITLE
Initial support for crusher.

### PR DESCRIPTION
**Pull request by joaander**
Implement the `install.sh` script for Crusher.

TODO:
- [x] Wait for HOOMD 3.9.0. release
- [x] Update HOOMD package to 3.9.0.
- [x] Wait for signac-flow release.
- [x] Update signac-flow.

### CODE REVIEW

**Comment by joaander** [comment]
cupy builds with experimental support, but doesn't work with hoomd:
```
HIP/ROCm does not support cuda array interface
```

**Comment by joaander** [comment]
Testing and working.

File: packages.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ fresnel:
 hoomd:
   ignore: beta,alpha,rc
   repository: https://github.com/glotzerlab/hoomd-blue
-  version: v3.8.1
+  version: fix-rocm-build
```
Comment (b-butler) [inline packages.yml]: Should we change this to trunk-minor now?

File: script/summit/install.sh [hunk 2]
```diff
@@ -147,13 +147,21 @@ fi
 
 
 
- git clone --recursive --branch v3.8.1 --depth 1 https://github.com/glotzerlab/hoomd-blue hoomd \
+
+
+if [ ! -n "$(ls -d $ROOT/lib/python*/site-packages/hoomd)" ]
+then
+
+ git clone --recursive --branch fix-rocm-build --depth 1 https://github.com/glotzerlab/hoomd-blue hoomd \
```
Comment (b-butler) [inline script/summit/install.sh]:
```suggestion
 git clone --recursive --branch trunk-minor --depth 1 https://github.com/glotzerlab/hoomd-blue hoomd \
```

File: template/crusher.jinja [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+#         -DCLANG_LINK_CLANG_DYLIB=ON \
+#         -DLLVM_BUILD_LLVM_DYLIB=ON \
+#         -DLLVM_LINK_LLVM_DYLIB=ON \
+#         -DCMAKE_BUILD_TYPE=Release \
+#         -DLLVM_TARGETS_TO_BUILD="PowerPC"
+#     cmake --build build -j {{ MAKEJOBS }}
+#     cmake --install build
+#     cd $BUILDDIR
+#     rm -rf llvm-project
+# fi
```
Comment (b-butler) [inline template/crusher.jinja]: Is there a reason for leaving this?

File: template/crusher_finalize.jinja [hunk 4]
```diff
@@ -0,0 +1,5 @@
+# cupy does not support the cuda array interface on crusher
+pip uninstall -y cupy
```
Comment (b-butler) [inline template/crusher_finalize.jinja:2]: Yep, CuPy is NVIDIA specific for now.

**Review by b-butler — CHANGES_REQUESTED** [review]
I hadn't notices the `make_dockerfiles.py` logic before. Its snazzy. I have a few questions and suggestions, but overall looks fine.

File: template/crusher_finalize.jinja (hunk 4 above)
Comment (joaander) [reply template/crusher_finalize.jinja:2]: CuPy builds for rocm successfully, but when the HOOMD unit tests use the `__cuda_array_interface__`, it gives an error:
https://docs.cupy.dev/en/stable/install.html#building-cupy-for-rocm-from-source
https://docs.cupy.dev/en/stable/install.html#limitations

File: template/crusher.jinja (hunk 3 above)
Comment (joaander) [reply template/crusher.jinja]: No.

File: packages.yml (hunk 1 above)
Comment (joaander) [reply packages.yml]: Yes, and then 3.9.0 after the hoomd 3.9 release and before merging this branch.

File: template/crusher_finalize.jinja (hunk 4 above)
Comment (b-butler) [reply template/crusher_finalize.jinja:2]: Oh that is news for me, they used to be CUDA only, but yeah not supporting `__cuda_array_interface__` is a non-starter.

**Review by b-butler — APPROVED** [review]

**Pull request merged**